### PR TITLE
Backend/group functions

### PR DESCRIPTION
**Pull request by danielsv03**
Added API routes for groups, POST, GET, DELETE.

### CODE REVIEW

File: Backend/Meta-TV2-api/Meta-TV2-AccessLayer/Controllers/ApiController.cs [hunk 1]
```diff
@@ -3,28 +3,43 @@ namespace Meta_TV2_api.Controllers;
...
+        return add ? Ok() : BadRequest("Failed to add group.");
+    }
 
+    [HttpGet]   
+    public async Task<IActionResult> GetGroups(){
+        var groups = await businessRules.GetGroups();
+        return groups != null ? Ok(groups) : NotFound("No groups found");
+    }
+
+    //Create a route for /Group/page=1&size=10
```
Comment (DanielNikolic03) [inline Backend/Meta-TV2-api/Meta-TV2-AccessLayer/Controllers/ApiController.cs:24]: Remove?

File: Backend/Meta-TV2-api/Meta-TV2-DataLayer/DataAccess.cs [hunk 2]
```diff
@@ -1,33 +1,20 @@
-﻿namespace Meta_TV2_DataLayer;
+﻿using Microsoft.EntityFrameworkCore;
+using Npgsql;
```
Comment (DanielNikolic03) [inline Backend/Meta-TV2-api/Meta-TV2-DataLayer/DataAccess.cs]: Remove

File: Backend/Meta-TV2-api/Meta-TV2-DataLayer/IDataAccess.cs [hunk 3]
```diff
@@ -2,35 +2,47 @@
...
+    /// <returns>Groups not archived</returns>
+    public Task<List<Groups>> GetGroups();
 
     /// <summary>
-    /// Test method for adding to the "Groups" table.
+    /// Function to get a group by id, that is not archived
     /// </summary>
-    /// <param name="group">Groups object to add</param>
-    public void Test_AddGroups(Groups group);
+    /// <param name="id"></param>
```
Comment (DanielNikolic03) [inline Backend/Meta-TV2-api/Meta-TV2-DataLayer/IDataAccess.cs:29]: Incomplete documentation

File: Backend/Meta-TV2-api/Meta-TV2-BusinessLayer/IBusinessRules.cs
Comment (DanielNikolic03) [inline Backend/Meta-TV2-api/Meta-TV2-BusinessLayer/IBusinessRules.cs:1]: No documentation

File: Backend/Meta-TV2-api/Meta-TV2-DataLayer/IDataAccess.cs [hunk 4]
```diff
@@ -2,35 +2,47 @@
 
 public interface IDataAccess
 {
+
+
+    /// <summary>
+    /// Test method for adding to the "Groups" table.
+    /// </summary>
+    /// <param name="group">Groups object to add</param>
+    public void AddGroups(Groups group);
```
Comment (DanielNikolic03) [inline Backend/Meta-TV2-api/Meta-TV2-DataLayer/IDataAccess.cs:11]: Interface is not updated. Implementation is async

Comment (danielsv03) [reply Backend/Meta-TV2-api/Meta-TV2-DataLayer/IDataAccess.cs:11]: Not necessary since, AddGroups is void

**Review by DanielNikolic03 — APPROVED** [review]
Cool

**Review by jonathan-blomlof — APPROVED** [review]

**Pull request merged**